packages/admin-api-schema/lib/index.js

```diff
@@ -47,39 +55,46 @@ const validate = ({data, version, schema, definitions}) => {
     return jsonSchema.validate(schemaJSON, definitionsJSON, data);
 };
 
-const augmentWithVersion = (version, fn, args) => {
-    return fn(Object.assign({}, {version}, args));
-};
-
-const v2 = {
-    get: (definitionName) => {
-        return augmentWithVersion('v2', get, {definitionName});
-    },
-    list: () => {
-        return augmentWithVersion('v2', list, {});
-    },
-    validate: (options) => {
-        return augmentWithVersion('v2', validate, options);
+const resolveVersion = (version) => {
+    if (version === 'v3') {
+        // NOTE: internally canary is used to identify latest available version, in this case it's 'v3'
+        //       once there's following version introduced - 'v4' that version should be substituted with 'canary'
+        return 'canary';
     }
-};
 
-const canary = {
-    get: (definitionName) => {
-        return augmentWithVersion('canary', get, {definitionName});
-    },
-    list: () => {
-        return augmentWithVersion('canary', list, {});
-    },
-    validate: (options) => {
-        return augmentWithVersion('canary', validate, options);
-    }
+    return version;
 };
 
+/**
+ * Versioned version of 'get' method, which returns a JSON schema definition if found
+ *
+ * @param {string} definitionName - JSON schema definitions name
+ * @param {string} version - API's JSON schema version to check against
+ *
+ * @returns {Object|null} JSON schema definition or null if it's not found
+ */
+const versionedGet = (definitionName, version = 'v3') => get({definitionName, version: resolveVersion(version)});
+
+/**
+ * Versioned version of 'list' method, which lisists available JSON schema definitions for provided version
+ *
+ * @param {string} version - API's JSON schema version to check against
+ *
+ * @returns {string[]} - list of available JSON schema definitions
+ */
+const versionedList = (version = 'v3') => list({version: resolveVersion(version)});
+
+/**
+ * Versioned version of 'validate' method, which validates objects against predefined JSON Schema
+ *
+ * @param {ValidateOptions} options
+ *
+ * @returns {Promise} - resolves a promise if validation is successful and rejects with error details otherwise
+ */
+const versionedValidate = options => validate(Object.assign(options, {version: resolveVersion(options.version || 'v3')}));
```

Comment:
Would be nice if this had the same pattern as get and list, like:
```js
/**
 * @param {ValidateOptions} options
 * @param {version='v3'} version
 *
 * @returns {Promise}
 */
```

Reply:
didn't find a proper syntax to reference ValidateOptions property an describe a default value :thinking:  Would expect something like this to work: ` * @param {ValidateOptions.version} [options.version=v3]`, but it didn't.

Reply:
Ah you would have to set the default in the typedef I think!

But my point of this comment was to have `version` as a separate argument, at the end, just like for `get` and `list`